src/main/scala/datalog/storage/CollectionsStorageManager.scala

```diff
@@ -69,6 +69,47 @@ abstract class CollectionsStorageManager(override val ns: NS) extends StorageMan
   def edbContains(rId: RelationId): Boolean = edbs.contains(rId)
   def getAllEDBS(): mutable.Map[RelationId, Any] = edbs.wrapped.asInstanceOf[mutable.Map[RelationId, Any]]
 
+  override def getAllPossibleEDBs(arity: Int): CollectionsEDB =
```

Comment:
For the report, it's worth discussing that this is truly needed and what other engines do (i.e. Souffle) since anytime something is combinatorial it's a pretty good indication that there may be an optimization somewhere.